src/platform/packages/shared/kbn-scout-upgrade-testing/test/discovery_tests/discovery.spec.ts

```diff
@@ -0,0 +1,165 @@
+    );
+    await kbnClient.uiSettings.update(defaultSettings);
+  });
+
+  test.beforeEach(async ({ browserAuth, pageObjects }) => {
+    await browserAuth.loginAsAdmin();
+    await pageObjects.discover.goto();
+  });
+
+  test.afterAll(async ({ kbnClient }) => {
+    await kbnClient.savedObjects.clean({ types: ['search', 'index-pattern'] });
+  });
+
+  test.describe('timepicker queries', () => {
+    const queryName1 = 'Query # 1';
+    const queryName2 = 'Query # 2';
+
+    test('should show correct time range string by timepicker', async ({ pageObjects }) => {
+      await pageObjects.timePicker.setDefaultAbsoluteRange();
+      const time = await pageObjects.timePicker.getTimeConfig();
+      expect(time.start).toBe(pageObjects.timePicker.defaultStartTime);
+      expect(time.end).toBe(pageObjects.timePicker.defaultEndTime);
+
+      const rowData = await pageObjects.discover.getDocTableIndex(1);
+      expect(rowData).toContain('Sep 22, 2015 @ 23:50:13.253');
+    });
+
+    test('save query should show toast message and display query name', async ({ pageObjects }) => {
+      await pageObjects.timePicker.setDefaultAbsoluteRange();
+      await pageObjects.discover.saveSearch(queryName1);
+      const actualQueryNameString = await pageObjects.discover.getCurrentQueryName();
+      expect(actualQueryNameString).toBe(queryName1);
+    });
+
+    test('should refetch when autofresh is enabled', async ({ pageObjects }) => {
+      const interval = 5;
+      await pageObjects.timePicker.setDefaultAbsoluteRange();
+      await pageObjects.timePicker.startAutoRefresh(interval);
+
+      const getRequestTimestamp = async () => {
+        await pageObjects.inspector.open();
+        const requestStats = await pageObjects.inspector.getTableData();
+        const requestStatsRow = requestStats.filter(
+          (r) => r && r[0] && r[0].includes('Request timestamp')
+        );
+        if (!requestStatsRow || !requestStatsRow[0] || !requestStatsRow[0][1]) {
+          await pageObjects.inspector.close();
+          return '';
+        }
+        await pageObjects.inspector.close();
+        return requestStatsRow[0][1];
+      };
+
+      const requestTimestampBefore = await getRequestTimestamp();
+
+      await expect
+        .poll(
+          async () => {
+            const requestTimestampAfter = await getRequestTimestamp();
+            return (
+              Boolean(requestTimestampAfter) && requestTimestampBefore !== requestTimestampAfter
+            );
+          },
+          { timeout: 30000 }
+        )
+        .toBe(true);
+    });
+
+    test('load query should show query name', async ({ pageObjects }) => {
+      await pageObjects.timePicker.setDefaultAbsoluteRange();
+      await pageObjects.discover.saveSearch(queryName2);
+      await pageObjects.discover.loadSavedSearch(queryName2);
+      await expect
+        .poll(async () => await pageObjects.discover.getCurrentQueryName())
+        .toBe(queryName2);
```

Comment:
same here, this functionality replicates FTR retry approach, which is something we should avoid in Scout.